### PR TITLE
[rv_core_ibex, sival] Extend rv_core_ibex testplan with SiVal tests

### PR DESCRIPTION
**Pull request by GregAC**
This moves the rv_core_ibex testplan into a seperate file, extends it with new tests and the extra SiVal test information fields. Additionally features have been added to the rv_core_ibex hjson data.

### CODE REVIEW

**Comment by GregAC** [comment]
Note I've put all the Ibex tests into SiVal stage 1. The reason for this is Ibex is obviously heavily used in all of the other tests, so this set of tests should be considered initial tests to check Ibex itself is working correctly before we test other things.

In particular we'll need a way to load/run these tests by injecting them into SRAM via JTAG, ideally before the mask ROM can execute at all (uncertain if we can actually halt that initial execution even on the DEV/UNLOCKED life cycle state)

**Review by rswarbrick — APPROVED** [review]
I think this looks sensible to me. I guess it probably makes sense to get this large "renaming commit" merged and then file followups to fix niggles?

**Comment by rswarbrick** [comment]
That said, CI looks like it's failing at the moment. Can you try running the quick lint locally:? It might show what's come unstuck!

File: hw/ip/rv_core_ibex/data/rv_core_ibex.hjson [hunk 1]
```diff
@@ -404,6 +404,68 @@
...
+    }
+    {
+      name: "RV_CORE_IBEX.ADDRESS_TRANSLATION"
+      descr: '''
+        The core wrapper implements address translation. RISC-V core memory accesses that match a
+        given range will have an offset applied before being sent to the memory system
+      '''
+    }
+    {
+      name: "RV_CORE_IBEX.CPU.INTERRUPTS"
```
Comment (marnovandermaas) [inline hw/ip/rv_core_ibex/data/rv_core_ibex.hjson:450]: Nit: put this with the other CPU features.

**Review by marnovandermaas — COMMENTED** [review]
It would be nice to have the moving of the test plan to be in a separate commit so that it is easier to review the text that you added.

**Review by HU90m — APPROVED** [review]
Testplan looks good

*For my future self:*
In terms of implementation, the following will likely be the least straight forward:
- `chip_sw_rv_core_ibex_icache_scrambled_access`
  - implemented using a vseq
- `chip_sw_rv_core_ibex_lockstep_glitch`
  - implemented using a vseq
- `chip_sw_rv_core_ibex_alerts`
  - involves injecting faults
- `chip_sw_rv_core_ibex_epmp_smoke`
  - Needs to be injected into SRAM using JTag

**Review by HU90m — COMMENTED** [review]
~~Are we sure the following are even possible to test on silicon, or should they be labeled None and left for pentesting?~~

- ~~`chip_sw_rv_core_ibex_icache_scrambled_access`~~
- ~~`chip_sw_rv_core_ibex_lockstep_glitch`~~
- ~~`chip_sw_rv_core_ibex_alerts`~~

Ignore me, they already are.

**Comment by rswarbrick** [comment]
CHANGE AUTHORIZED: hw/ip/rv_core_ibex/data/rv_core_ibex.hjson

**Comment by a-will** [comment]
CHANGE AUTHORIZED: hw/ip/rv_core_ibex/data/rv_core_ibex.hjson

**Pull request merged**